openff/evaluator/utils/openmm.py

```diff
@@ -179,12 +179,30 @@ def system_subset(
 
     handlers_to_register = {parameter_key.tag}
 
-    if parameter_key.tag in {"ChargeIncrementModel", "LibraryCharges"}:
-        # Make sure to retain all of the electrostatic handlers when dealing with
-        # charges as the applied charges will depend on which charges have been applied
-        # by previous handlers.
+    if parameter_key.tag in {"ChargeIncrementModel", "LibraryCharges", "VirtualSites"}:
+        # Make sure to retain _all_ of the electrostatic handlers when dealing with any handler
+        # that might modify changes as the applied charges will depend on which charges have been
+        # applied by previous handlers.
         handlers_to_register.update(
-            {"Electrostatics", "ChargeIncrementModel", "LibraryCharges"}
+            {
+                "Electrostatics",
+                "ChargeIncrementModel",
+                "LibraryCharges",
+                "VirtualSiteHandler",
+                "ToolkitAM1BCC",
+            }
+        )
+
+    if parameter_key.tag in {"VirtualSites"}:
+        # Interchange's current implementation uses bonds and constraints to determine the values
+        # OpenMM needs for virtual sites; using positions might not produce accurate results since a
+        # conformer's geometry likely does not match the force field geometry
+        handlers_to_register.update(
+            {
+                "vdW",
```

Comment:
I don't think we need the vdW handler if we take the gradient of an electrostatic parameter, I am not sure how much of a performance difference this would actually make however.

Reply:
Ahhh no my bad I forgot we allow v-sites to have vdw interactions as well so I think you were correct sorry about that!

Reply:
I was a little confused - IIUC `vdWHandler` and `VirtualSiteHandler` completely separate the interactions, i.e. one handles all of the vdW interactions on the atoms and the other only on the virtual sites, no interaction or cross-talk like happens with charge increments, etc. in electrostatics. But then I guess you actually need the atomic vdW to evaluate the virtual sites' vdW since the system subset is slicing the system by interaction - in other words, you'd need all vdW interactions whether or not you're fitting all of them.

Is this coherent at all? I think we do need `"vdW"` included in this block, so if you also do, it's okay if I'm not making sense here.

Reply:
Exactly we need to include all handlers which might touch that interaction we are fitting to and as our v-sites can hit both vdW and electrostatics we need to be careful to include them. So I agree they should be in this block. 

We don't actually use v-sites with vdW parameters yet but if we do I can see the reverse also being an issue. Where we fit vdW and include v-sites with charges and LJ terms, we would only have charges on the atoms with v-sites and nothing else.